corehq/apps/locations/views.py

```diff
@@ -336,7 +337,14 @@ def mk_loctype(name, parent_type, administrative,
 
         for loc_type in hierarchy:
             # make all locations in order
-            mk_loctype(**loc_type)
+            try:
+                mk_loctype(**loc_type)
+            except IntegrityError:
```

Comment:
I'm not sure about this, as this could conceivably arise by other means.

There's a bit of code in this view that's supposed to detect this:  https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/locations/views.py#L327-L330 The problem is it doesn't work if the codes are inferred.  You could fix this by generating the code earlier rather than just not validating autogenerated codes.

Even better, this could avoid conflicts entirely (for autogenerated codes) by making sure the generated code is unique, for instance, using https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/commtrack/util.py#L255-L255.